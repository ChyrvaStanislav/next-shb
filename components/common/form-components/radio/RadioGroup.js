import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { ErrorMessage } from 'formik';
import HintIcon from 'v2_main/components/common/form-components/hint-icon';
import RadioInput from 'v2_main/components/common/form-components/radio';
import { Typography } from 'v2_main/components/html';
import styles from './RadioGroup.module.scss';

const RadioGroup = (props) => {
  const {
    data, name, label, selectedNoLabel, toShowNoLabel, required, formId, icon, selectedValue, form, ...otherProps
  } = props;

  return (
    <section className={styles.radioGroupContainer}>
      <div>
        <label
          className={classNames(styles.label, {
            [styles.required]: required
          })}
        >
          {label}
        </label>
        {icon && <HintIcon />}
      </div>
      <div className={styles.radioContainer}>
        {
          data.map(item => (
            <RadioInput
              {...otherProps}
              name={name}
              id={item}
              value={item}
              key={item}
              label={item}
              formId={formId}
              selectedvalue={selectedValue}
              error={form?.errors?.[name] && form?.touched?.[name]}
            />
          ))
        }
      </div>
      {toShowNoLabel && selectedValue === 'No' && (
        <div className={classNames(styles.label, styles.checkedNoLabel)}>
          <div
            className={styles.iconWrapper}
          >
            <div className={styles.icon}>
              !
            </div>
          </div>
          <span>
            {selectedNoLabel}
          </span>
        </div>
      )}
      <Typography
        type="span"
        variant="small_forms"
      >
        <ErrorMessage name={name} component="span" className={styles.error} />
      </Typography>
    </section>
  );
};

RadioGroup.propTypes = {
  formId: PropTypes.string.isRequired,
  data: PropTypes.arrayOf(PropTypes.string).isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  selectedNoLabel: PropTypes.string,
  required: PropTypes.bool,
  icon: PropTypes.bool,
  selectedValue: PropTypes.string,
  toShowNoLabel: PropTypes.bool,
  form: PropTypes.shape({
    errors: PropTypes.object,
    touched: PropTypes.object,
  })
};

RadioGroup.defaultProps = {
  label: '',
  selectedNoLabel: '',
  required: false,
  icon: false,
  selectedValue: '',
  toShowNoLabel: false,
  form: {
    errors: {},
    touched: {},
  }
};

export default RadioGroup;
