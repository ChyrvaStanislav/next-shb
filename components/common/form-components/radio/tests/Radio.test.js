import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { Formik } from 'formik';
import RadioGroup from '../RadioGroup';

const Component = props => (
  <Formik>
    <RadioGroup
      data={['Yes', 'No']}
      name="commercialUsage"
      label="I plan to use this solution for my business, rather than academic or personal needs"
      selectedNoLabel="Please note: if you leave this solution for commercial use, you will receive support from EPAM."
      toShowNoLabel={true}
      icon={true}
      formId="formId"
      required={true}
      selectedValue="Yes"
      {...props}
    />
  </Formik>
);

describe('RadioGroup renders', () => {
  it('should RadioGroup renders', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should RadioGroup with selected value "No" renders', () => {
    const { asFragment } = render(<Component selectedValue="No" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Default selected', () => {
  it('should default selected works', () => {
    render(<Component />);
    const input = document.querySelector('#formId-Yes');
    expect(input.checked).toBeTruthy();
  });

  it('should default non-selected works', () => {
    render(<Component />);
    const input = document.querySelector('#formId-No');
    expect(input.checked).not.toBeTruthy();
  });
});

describe('Checked on "click"', () => {
  it('should checked/unchecked on "click"', () => {
    render(<Component />);
    const input = document.querySelector('#formId-Yes');
    waitFor(() => fireEvent.click(input)).finally(() => {
      expect(input.checked).not.toBeTruthy();
      waitFor(() => fireEvent.click(input)).finally(() => {
        expect(input.checked).toBeTruthy();
      });
    });
  });
});
