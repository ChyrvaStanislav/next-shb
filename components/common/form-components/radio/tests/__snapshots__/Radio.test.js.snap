// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders should RadioGroup renders 1`] = `
<DocumentFragment>
  <section
    class="radioGroupContainer"
  >
    <div>
      <label
        class="label required"
      >
        I plan to use this solution for my business, rather than academic or personal needs
      </label>
      <section
        class="wrapperComponent"
        data-testid="popup"
        role="presentation"
        tabindex="1"
      >
        <div
          class="targetElement"
        >
          <div
            class="iconWrapper"
            data-testid="targetDiv"
          >
            <div
              class="icon"
            >
              i
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="radioContainer"
    >
      <section
        class="radioInputContainer"
      >
        <label
          class="label"
          for="formId-Yes"
        >
          <input
            checked=""
            class="input"
            id="formId-Yes"
            name="commercialUsage"
            selectedvalue="Yes"
            type="radio"
            value="Yes"
          />
          <span
            class="radioInput"
          />
          Yes
        </label>
      </section>
      <section
        class="radioInputContainer"
      >
        <label
          class="label"
          for="formId-No"
        >
          <input
            class="input"
            id="formId-No"
            name="commercialUsage"
            selectedvalue="Yes"
            type="radio"
            value="No"
          />
          <span
            class="radioInput"
          />
          No
        </label>
      </section>
    </div>
    <span
      class="small_forms"
      other=""
    />
  </section>
</DocumentFragment>
`;

exports[`RadioGroup renders should RadioGroup with selected value "No" renders 1`] = `
<DocumentFragment>
  <section
    class="radioGroupContainer"
  >
    <div>
      <label
        class="label required"
      >
        I plan to use this solution for my business, rather than academic or personal needs
      </label>
      <section
        class="wrapperComponent"
        data-testid="popup"
        role="presentation"
        tabindex="1"
      >
        <div
          class="targetElement"
        >
          <div
            class="iconWrapper"
            data-testid="targetDiv"
          >
            <div
              class="icon"
            >
              i
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="radioContainer"
    >
      <section
        class="radioInputContainer"
      >
        <label
          class="label"
          for="formId-Yes"
        >
          <input
            class="input"
            id="formId-Yes"
            name="commercialUsage"
            selectedvalue="No"
            type="radio"
            value="Yes"
          />
          <span
            class="radioInput"
          />
          Yes
        </label>
      </section>
      <section
        class="radioInputContainer"
      >
        <label
          class="label"
          for="formId-No"
        >
          <input
            checked=""
            class="input"
            id="formId-No"
            name="commercialUsage"
            selectedvalue="No"
            type="radio"
            value="No"
          />
          <span
            class="radioInput"
          />
          No
        </label>
      </section>
    </div>
    <div
      class="label checkedNoLabel"
    >
      <div
        class="iconWrapper"
      >
        <div
          class="icon"
        >
          !
        </div>
      </div>
      <span>
        Please note: if you leave this solution for commercial use, you will receive support from EPAM.
      </span>
    </div>
    <span
      class="small_forms"
      other=""
    />
  </section>
</DocumentFragment>
`;
